Angular/src/components/text-based-gemini/text-based-gemini.component.ts

```diff
@@ -23,25 +22,12 @@ export class TextBasedGeminiComponent implements OnInit, OnDestroy {
    * @param text
    */
   askGemini(text: string) {
-    // this.subscription = this.genAiService.askGemini(text).subscribe({
-    //   next: (response: string) => {
-    //     console.log(response);
-    //     alert(`Response from Gemini: ${response}`);
-    //   },
-    //   error: (error: any) => {
-    //     console.error('Error:', error);
-    //     alert('An error occurred while fetching the response from Gemini.');
-    //   },
-    //   complete: () => {
-    //     console.log('Gemini response completed successfully.');
-    //   }
-    // });
-  }
-
-  ngOnDestroy(): void {
-    // Unsubscribe from the Observable if it exists
-    if (this.subscription) {
-      this.subscription.unsubscribe();
-    }
+    this.genAiService.askGemini(text).then(
+      (result: string) => {
+        this.response.set(result);
+      },
+      (error: Error) => {
+        alert(error);
```

Comment:
We should use something else than an alert. I assume this is just temporary?

Reply:
yes obviously we will consume a npm of `toastify `later